pandas/_libs/lib.pyx

```diff
@@ -2035,10 +2035,58 @@ cdef class IntervalValidator(Validator):
 
 
 cpdef bint is_interval_array(ndarray values):
+    """
+    Is this an ndarray of Interval (or np.nan) with a single dtype?
+    """
+
     cdef:
-        IntervalValidator validator = IntervalValidator(len(values),
-                                                        skipna=True)
-    return validator.validate(values)
```

Comment:
can you remove the IntervalValidator code?

Reply:
sure, will add that to the upcoming misc branch

Reply:
kk